docs/source/generics.rst

```diff
@@ -0,0 +1,986 @@
+Generics
+========
+
+You may have seen type hints like ``list[str]`` or ``dict[str, int]`` in Python
```

Comment:
General question: I'm not sure how backwards-compatible these docs are meant to be. In your last PR, I noticed you used `Optional[X]` instead of `X | None` throughout, but didn't mention it because I assumed you were trying to keep the code examples compatible with py38+. But this document uses PEP-585 syntax throughout, and doesn't mention the older way of writing these types until the last section right at the bottom.

I prefer making the new syntax more prominent than the old syntax, as this document does, but maybe we should have the note about how to do things in a pre-PEP-585 world slightly higher up?

Reply:
I think the ideal would be if we had some selector at the top of examples, similar to the one for Unix/Windows at https://packaging.python.org/en/latest/tutorials/packaging-projects/, where users could toggle between "Python 3.9+" and "Python 3.7-3.8" or so.

Setting that up is better left for a separate PR though.

Reply:
Yes, that would be ideal!

Reply:
For the mypy docs this is based on, I went by https://pypistats.org/packages/mypy minor version counts to determine what version should be emphasised. 3.9 is mode and median.